main.js

```diff
@@ -385,72 +413,75 @@ const searchFestival = async (searchTerm) => {
+    );
+  }
 };
 
 //#endregion
 //-> adding event listeners to the search field
 //#region
-function searchByInput () {
-    const searchInput = document.getElementById("search-input");
-    let searchTerm = "";
-
-    //recording input
-    searchInput.addEventListener("change", (event) => {
-        searchTerm = event.target.value;
-    });
-
-    //fetching data according to search word after uses presses enter
-    searchInput.addEventListener("keyup", (event) => {
-        if (event.key === "Enter") {
-            searchFestival(searchTerm);
-        };
-    });
-};
+function searchByInput() {
+  const searchInput = document.getElementById("search-input");
+  let searchTerm = "";
+
+  //recording input
+  searchInput.addEventListener("change", (event) => {
+    searchTerm = event.target.value;
+  });
+
+  //fetching data according to search word after uses presses enter
+  searchInput.addEventListener("keyup", (event) => {
+    if (event.key === "Enter") {
+      searchFestival(searchTerm);
+    }
+  });
+}
 
 //#endregion
 //-> see all results functionality for the dynamic search (page reload)
 //#region
-function addAllResultsButton () {
-    const button = document.querySelector("#all-results-button");
+function addAllResultsButton() {
+  const button = document.querySelector("#all-results-button");
 
-    const refreshPage = () => {
+  const refreshPage = () => {
     location.reload();
-    }; 
+  };
 
-    button.addEventListener('click', refreshPage);
-    button.classList.remove("d-none");
-};
+  button.addEventListener("click", refreshPage);
+  button.classList.remove("d-none");
+}
 
 //#endregion
 //-> search and filter clearing functions
 //#region
 //clear the search fields
-function clearDynamicSearch () {
-
-    if (document.getElementById("search-input") != null) {
-        document.getElementById("search-input").value = "";
-    };
-    if (document.getElementById("table-body") != null) {
-        document.getElementById("table-body").innerHTML = "";
-    }; 
-};
+// ! good that you are cleaning the search input!
+function clearDynamicSearch() {
+  if (document.getElementById("search-input") != null) {
+    document.getElementById("search-input").value = "";
+  }
+  if (document.getElementById("table-body") != null) {
+    document.getElementById("table-body").innerHTML = "";
+  }
+}
 
 //clearing the filter input fields and values
-function clearFilters () {
-    const neighbourhoodSelect = document.getElementById("neighbourhoodSelect")
-
-    if (neighbourhoodSelect != null) {
-        neighbourhoodSelect.innerHTML = "";
-        const allOption = document.createElement("option");
-        allOption.value = "all";
-        allOption.innerText = "All";
-        neighbourhoodSelect.appendChild(allOption);
-
-        document.getElementById("monthInput").value = "";
-    }; 
-};
+// ! looks like here you're just adding the "All" option to the select. you could have created this option statically in html directly
```

Comment:
This is a function that clears the entire inner html to remove all the options - this way the filter is ready to create new options from the dynamic search. Thank you for pointing this out, I found a nicer way of doing this (required adding id to the all option): 
```
while (neighbourhoodSelect.lastChild.id != "all-option") {
        neighbourhoodSelect.removeChild(neighbourhoodSelect.lastChild);
    };
```